plugin/trino-hive-hadoop2/src/test/java/io/trino/plugin/hive/HadoopQueryRunner.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.net.HostAndPort;
+import io.airlift.log.Logger;
+import io.airlift.log.Logging;
+import io.trino.plugin.hive.authentication.NoHdfsAuthentication;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.plugin.hive.metastore.MetastoreConfig;
+import io.trino.plugin.hive.metastore.thrift.BridgingHiveMetastore;
+import io.trino.plugin.hive.metastore.thrift.DefaultThriftMetastoreClientFactory;
+import io.trino.plugin.hive.metastore.thrift.NoHiveMetastoreAuthentication;
+import io.trino.plugin.hive.metastore.thrift.StaticMetastoreConfig;
+import io.trino.plugin.hive.metastore.thrift.StaticMetastoreLocator;
+import io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastore;
+import io.trino.plugin.hive.metastore.thrift.ThriftMetastoreAuthenticationConfig;
+import io.trino.plugin.hive.metastore.thrift.ThriftMetastoreConfig;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.TestingNodeManager;
+import io.trino.tpch.TpchTable;
+
+import java.util.Map;
+import java.util.function.Function;
+
+public final class HadoopQueryRunner
+{
+    private HadoopQueryRunner() {}
+
```

Comment:
this new thing is super cool. let's make sure it works.

let's run ConnectorSmokeTest with the new runner.